crd/hack/update-codegen.sh

```diff
@@ -34,98 +34,26 @@ export GO111MODULE GOFLAGS GOPATH
-  crd_name="$1"
-  version="$2"
-  crd_version="$3"
-  apis_pkg="${APIS_BASE_PKG}/${1}/${2}/..."
 
-  if [[ $# -eq 4 ]]; then
-    apis_pkg="{${apis_pkg},${APIS_BASE_PKG}/${1}/${4}/...}"
-  fi
-
-  echo "Performing code generation for ${crd_name} CRD"
-  echo "Generating deepcopy functions and CRD artifacts"
-  go run sigs.k8s.io/controller-tools/cmd/controller-gen \
-          object:headerFile="${SCRIPT_ROOT}/hack/boilerplate.go.txt" \
-          crd:crdVersions="${crd_version}" \
-          paths="${apis_pkg}" \
-          output:crd:artifacts:config="${SCRIPT_ROOT}/config/crds"
-}
-
-codegen_for () {
-  local crd_name version crd_version extra_version apis_pkg output_pkg
-
-  if [[ $# -ne 3 ]] && [[ $# -ne 4 ]]; then
-    echo "Usage: codegen_for CRD-NAME VERSION CRDSPEC-VERSION optional:EXTRA-VERSION" >&2
-    echo "" >&2
-    echo "This writes auto generated client methods for CRD-NAME/VERSIONs" >&2
-    return 1
-  fi
-
-  crd_name="$1"
-  version="$2"
-  crd_version="$3"
-  output_pkg="${OUTPUT_BASE_PKG}/${1}"
-  apis_pkg="${APIS_BASE_PKG}/${1}/${2}"
-
-  if [[ $# -eq 4 ]]; then
-    extra_version="$4"
-    apis_pkg="${apis_pkg},${APIS_BASE_PKG}/${1}/${extra_version}"
-    generate_config "$crd_name" "$version" "$crd_version" "$extra_version"
-  else
-    generate_config "$crd_name" "$version" "$crd_version"
-  fi
-
-
-  echo "Generating clientset at ${output_pkg}/${CLIENTSET_PKG_NAME}"
-  echo "apis_pkg ${apis_pkg}"
-  echo "output_pkg/CLIENTSET_PKG_NAME ${output_pkg}/${CLIENTSET_PKG_NAME}"
-  go run k8s.io/code-generator/cmd/client-gen \
-          --input-base "" \
-          --input "${apis_pkg}" \
-          --clientset-name "${CLIENTSET_NAME}" \
-          --output-package "${output_pkg}/${CLIENTSET_PKG_NAME}" \
-          ${COMMON_FLAGS}
-
-  echo "Generating listers at ${output_pkg}/listers"
-  go run k8s.io/code-generator/cmd/lister-gen \
-          --input-dirs "${apis_pkg}" \
-          --output-package "${output_pkg}/listers" \
-          ${COMMON_FLAGS}
-
-  echo "Generating informers at ${output_pkg}/informers"
-  go run k8s.io/code-generator/cmd/informer-gen \
-           --input-dirs "${apis_pkg}" \
-           --versioned-clientset-package "${output_pkg}/${CLIENTSET_PKG_NAME}/${CLIENTSET_NAME}" \
-           --listers-package "${output_pkg}/listers" \
-           --output-package "${output_pkg}/informers" \
-           ${COMMON_FLAGS}
-
-  echo "Generating register at ${apis_pkg}"
-  go run k8s.io/code-generator/cmd/register-gen \
-          --input-dirs "${apis_pkg}" \
-          --output-package "${apis_pkg}" \
-          ${COMMON_FLAGS}
-}
-
-codegen_for network v1alpha1 v1 v1
-codegen_for gcpfirewall v1beta1 v1
+echo "Generating CRD artifacts"
+go run sigs.k8s.io/controller-tools/cmd/controller-gen crd \
+        object:headerFile="${SCRIPT_ROOT}/hack/boilerplate.go.txt" \
+        paths="${SCRIPT_ROOT}/apis/..." \
+        output:crd:artifacts:config="${SCRIPT_ROOT}/config/crds"
```

Comment:
This is complaining about the lack of a newline.

Reply:
pushed an extra commit to fix this